src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -671,9 +671,28 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
             $dumper->setProxyDumper(new ProxyDumper(md5($cache->getPath())));
         }
 
-        $content = $dumper->dump(array('class' => $class, 'base_class' => $baseClass, 'file' => $cache->getPath(), 'debug' => $this->debug));
+        $content = $dumper->dump(array('class' => $class, 'base_class' => $baseClass, 'file' => $cache->getPath(), 'as_files' => !$this->debug, 'debug' => $this->debug));
```

Comment:
I haven't reviewed this PR, but out of context, yea, it looks weird to switch things based on the debug flag. What's the idea?